### PR TITLE
Dm/icon page 2016 10 10

### PR DESCRIPTION
**Pull request by davidpmccormick**

### CODE REVIEW

File: client/images/icons/svg-sprite.njk [hunk 1]
```diff
@@ -1,3 +1,3 @@
-<div style="height: 0; width: 0; position: absolute; visibility: hidden">
+<div id="icon-sprite" style="height: 0; width: 0; position: absolute; visibility: hidden">
```
Comment (jamesgorrie) [inline client/images/icons/svg-sprite.njk]: Why not have this styling in the stylesheets?

File: client/js/components/icons.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    iconHeading.appendChild(document.createTextNode(iconId));
+    iconHeading.setAttribute('class', 'styleguide__item-subheading');
+
+    iconWrapper.appendChild(iconHeading);
+    useEl.setAttributeNS('http://www.w3.org/1999/xlink', 'xlink:href', '#' + iconId);
+    svgEl.appendChild(useEl);
+    iconWrapper.appendChild(svgEl);
+    el.appendChild(iconWrapper);
+  }
+};
```
Comment (jamesgorrie) [inline client/js/components/icons.js]: I wonder if there is someway we can add the vars to the server, then push them out to the sass and templates... Rule of three though.

Comment (jamesgorrie) [inline client/js/components/icons.js]: Also thinking of using something like `v-dom` for dom manipulation - it's way easier to fathom what's going on.

File: client/js/components/icons.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+export default (el) => {
+  const iconSpriteEl = document.getElementById('icon-sprite');
+  const icons = iconSpriteEl.getElementsByTagName('symbol');
+
+  for (let i = 0, l = icons.length; i < l; i++) {
```
Comment (jamesgorrie) [inline client/js/components/icons.js]: Could we just use a `forEach` here?

Comment (davidpmccormick) [inline client/js/components/icons.js]: Yep – updated.

File: client/images/icons/svg-sprite.njk (hunk 1 above)
Comment (davidpmccormick) [inline client/images/icons/svg-sprite.njk]: Good point – thanks.

File: client/js/components/icons.js (hunk 2 above)
Comment (davidpmccormick) [inline client/js/components/icons.js]: Rule of three?

I can confess to feeling slightly naked without jQuery to fall back on for anything DOM-related, but I'm keen only to use what we need to get by for perf reasons (and personal development).

Comment (davidpmccormick) [inline client/js/components/icons.js]: Aha. Not [rule of three](https://en.wikipedia.org/wiki/Rule_of_three_(writing)). Right.

**Pull request merged**